tensorflow/core/kernels/ragged_range_op.cc

```diff
@@ -91,12 +91,12 @@ class RaggedRangeOp : public OpKernel {
                             Eigen::numext::abs(delta));
       } else {
         // The following is copied from tensorflow::RangeOp::Compute().
-        auto size_auto =
-            Eigen::numext::ceil(Eigen::numext::abs((limit - start) / delta));
-        OP_REQUIRES(
-            context, size_auto <= std::numeric_limits<int64_t>::max(),
-            errors::InvalidArgument("Requires ((limit - start) / delta) <= ",
-                                    std::numeric_limits<int64_t>::max()));
+        auto size_auto = Eigen::numext::ceil(
+            Eigen::numext::abs((limit / delta) - (start / delta)));
```

Comment:
This does not produce the same results as before due to truncation.  For example,
```python

start = 1136033461
end = -2110457152
step = -3

count1 = (end - start) // step
count2 = end // step - start // step

print(count1, count2)
```
produces:
```python
1082163537 1082163538
```

Reply:
To avoid overflow for integers, you will need to take the difference in an unsigned type.

Reply:
> This does not produce the same results as before due to truncation.  For example,

Ah that makes sense

Reply:
> To avoid overflow for integers, you will need to take the difference in an unsigned type.

This is not going to fix overflow, it just avoids it, right?

What if, during the calculations, we cast to a wider type, and after the division by delta, we cast it back to it's original type

Reply:
It will fix overflow.  For example, if `T` is `int64_t`, then
```
uint64_t range = static_cast<uint64_t>(limit - start);
```
will be correct, as long as `limit > start`.  You can use `std::make_unsigned_t` to get an unsigned version of an integer.

There may not be a wider type, e.g. for `int64_t`.  And casting back may not help if `delta` is small, like `+/-1`.